reconcile/utils/glitchtip/models.py

```diff
@@ -35,25 +38,31 @@ class Team(BaseModel):
     users: list[User] = []
 
     @root_validator(pre=True)
-    def name_xor_slug_must_be_set(cls, values):  # pylint: disable=no-self-argument
+    def name_xor_slug_must_be_set(  # pylint: disable=no-self-argument
+        cls, values: MutableMapping[str, Any]
+    ) -> MutableMapping[str, Any]:
```

Comment:
Sorry if this isn't clear in the docs, but the intent is the abstract/protocol types in inputs and the concrete types in the outputs (see [here](https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-style-guide/#type-hints)).

So, `Mapping` and `MutableMapping` for inputs, but `dict` is fine for the return value.

Reply:
I tried to have the return type as `dict[str, Any]`, but mypy doesn't like it. 

Take this easy example:

```python
def foobar(values: Mapping[str, str]) -> dict[str, str]:
    return values
```

And mypy complains:

```
error: Incompatible return value type (got "Mapping[str, str]", expected "Dict[str, str]")
```

Sorry, my type hints kungfu is weak today :(

Reply:
Sorry, I should have paid more attention to what was going on in these functions. This is a common error when we're modifying some mutable object and returning it. Normally we could probably skip returning a mutable object from a function because we have the original reference to it anyways, but this looks like a pydantic implementation detail, so for these sorts of cases using `Mapping` or `MutableMapping` for both the input and return makes sense.

Thanks again for the work here!